src/parser/mage/fire/modules/features/CombustionFirestarter.tsx

```diff
@@ -19,40 +19,38 @@ const FIRESTARTER_HEALTH_THRESHOLD = .90;
 const debug = false;
 
 class CombustionFirestarter extends Analyzer {
+
   combustionCast = false;
-  combustionCastEvent = null;
   combustionDuringFirestarter = false;
   healthPercent = 1;
 
-  constructor(...args) {
-    super(...args);
+  constructor(options: any) {
+    super(options);
     this.active = this.selectedCombatant.hasTalent(SPELLS.FIRESTARTER_TALENT.id);
     this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.COMBUSTION), this.onCombustion);
     this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(DAMAGE_SPELLS), this.onDamage);
   }
 
   //Checks to see if a new Combustion was cast. This variable is marked false once a damage event is triggered since we only want the first damage event in the Combustion (to get the health percentage)
-  onCombustion(event) {
+  protected combustionCastEvent: ApplyBuffEvent | undefined;
+  onCombustion(event: ApplyBuffEvent) {
     this.combustionCast = true;
     this.combustionCastEvent = event;
```

Comment:
I think this is unused?

Reply:
yea, i removed the timeline stuff that used to be in there cause i realized i was using the event from an applybuff event instead of a cast event and missed that line.

Thought about changing the apply buff event to a cast event instead so timeline would work, but i dont quite remember why i chose to use apply buff instead of cast. I think it has to do with if someone hits combustion prepull, but i just removed the timeline cause i dont want to break anything.